### ISSUE TITLE
Add Save + Restore

### ISSUE THREAD

**Issue by laurenzfg** [laurenzfg/notenstreicher#1]
The user should be able to save and load a transcript in the UI.

**Comment by septatrix** [comment]
Do you mean like parsing the export from RWTHonline or a simple store in the browser for the next visit?

**Comment by laurenzfg** [comment]
I meant the latter, but the former would be exceptionally cool. But idk if RWTHOnline has a structured export (neither scraping the Web App nor parsing the transcript PDF seems to be an easy endeavor).
If you have capacity to work on any of these features, feel encouraged to contribute a PR.

**Comment by septatrix** [comment]
I am currently working on my own version of this (and experimenting a bit with Svelte, Rust+WASM while doing so) and actually got the PDF extraction working. For some weird reason though it does not work after bundling for production, only when using npm serve :thinking: Otherwise it reads everything as the empty string :/

**Comment by laurenzfg** [comment]
I implemented and deployed local storage for the transcript, including an opt-out facility. I believe including an import from RWTH Online would be very nice.

@septatrix did you manage to make it work? Can you maybe share the component with me? If it is a Rust WASM, we should be able to bundle it into this web application.

**Comment by septatrix** [comment]
> @septatrix did you manage to make it work? Can you maybe share the component with me? If it is a Rust WASM, we should be able to bundle it into this web application.

Yes, you can find it here: https://septatrix.github.io/grater/. The PDF extraction code is written in JS and uses PDF.js to parse the PDF.

My initial plan was to move everything to rust though the extraction code is in large parts adopted from a different project and I do not yet fully understand it / want to simplify some parts of it. Admittedly I have not worked on it for a while - and I know of a few errors and improvements which are still do be done.

Also an advantage of doing the parsing using PDF.js is that it would be simple to show to the user which courses/grades were detected by showing the PDF and marking the detected text so I am not even sure if I really want to move the extraction to rust...
